reader/src/main/java/org/jline/reader/impl/LineReaderImpl.java

```diff
@@ -3431,24 +3440,9 @@ private AttributedString getDisplayedBufferWithPrompts(List<AttributedString> se
         return full.toAttributedString();
     }
 
-    private AttributedString getMaskedBuffer(String buffer) {
-        if (mask != null) {
-            if (mask == NULL_MASK) {
-                buffer = "";
-            } else {
-                StringBuilder sb = new StringBuilder();
-                for (int i = buffer.length(); i-- > 0;) {
-                    sb.append((char) mask);
-                }
-                buffer = sb.toString();
-            }
-        }
-        return new AttributedString(buffer);
-    }
-
     private AttributedString getHighlightedBuffer(String buffer) {
-        if (mask != null) {
-            return getMaskedBuffer(buffer);
+        if (lineReaderCallback != null) {
```

Comment:
I don't like this piece of logic much.  In the previous logic, the mask was used everywhere, so it clearly did not make much sense to use a highlighter with a fully masked input.  However, with this new feature, I think the masking should be done without forbidding highlighting.
Maybe have `onHighlightLine` return a `String` which would be given to the highlighter if configured ?